addon/modifiers/au-floating-ui.js

```diff
@@ -0,0 +1,158 @@
+import Modifier from 'ember-modifier';
+import { assert } from '@ember/debug';
+import { registerDestructor } from '@ember/destroyable';
+
+import {
+  autoUpdate,
+  computePosition,
+  flip,
+  hide,
+  offset,
+  arrow,
+} from '@floating-ui/dom';
+
+import { merge } from 'merge-anything';
+
+export default class AuFloatingUiModifier extends Modifier {
+  modify(
+    floatingElement,
+    [_referenceElement, _arrowElement],
+    { defaultPlacement = 'bottom-start', options = {}, middleware = [] }
+  ) {
+    const referenceElement =
+      typeof _referenceElement === 'string'
+        ? document.querySelector(_referenceElement)
+        : _referenceElement;
+
+    const arrowElement =
+      typeof _arrowElement === 'string'
+        ? document.querySelector(_arrowElement)
+        : _arrowElement;
+
+    const defaultOptions = {
+      floater: {
+        offset: 6,
+      },
+      arrow: {
+        offset: 4,
+        padding: 3,
+        position: 'min(15%, 12px)',
+      },
+    };
+    options = merge(defaultOptions, options);
+
+    assert(
+      `AuFloatingUI (modifier): No reference element was defined.`,
+      referenceElement instanceof HTMLElement
+    );
+
+    assert(
+      `AuFloatingUI (modifier): No floating element was defined.`,
+      floatingElement instanceof HTMLElement
+    );
+
+    assert(
+      `AuFloatingUI (modifier): The reference and floating elements cannot be the same element.`,
+      floatingElement !== _referenceElement
+    );
+
+    assert(
+      `AuFloatingUI (modifier): @placement must start with either 'bottom' or 'top'.`,
```

Comment:
I only provided support for placement of the floater on the `top` and `bottom` side of the reference element. I presume this is enough for now? If needed, this can be easily extended to support `right` and `left`.